sw/host/opentitantool/src/command/mod.rs

```diff
@@ -21,24 +21,45 @@ pub mod transport;
 pub mod update_usr_access;
 pub mod version;
 
-use anyhow::Result;
+use anyhow::{bail, Result};
+use regex::Regex;
 use serde_annotate::Annotate;
 use std::any::Any;
+use std::time::Duration;
 use structopt::StructOpt;
 
 use opentitanlib::app::command::CommandDispatch;
 use opentitanlib::app::TransportWrapper;
 
+fn parse_duration(s: &str) -> Result<Duration> {
+    let re = Regex::new(r"^([0-9]+) *([a-zA-Z]+)$").unwrap();
+    let Some(captures) = re.captures(s) else {
+        bail!("Unrecognized time format, try `3s` or `500ms`")
+    };
+    let number: u64 = captures[1].parse()?;
+    match captures[2].as_ref() {
+        "ms" => Ok(Duration::from_millis(number)),
+        "s" => Ok(Duration::from_secs(number)),
+        _ => bail!("Unrecognized time format, try `3s` or `500ms`"),
+    }
+}
+
 #[derive(Debug, StructOpt)]
 /// No Operation.
-pub struct NoOp {}
+pub struct NoOp {
+    #[structopt(short = "d", long, help = "Delay execution", parse(try_from_str = parse_duration))]
```

Comment:
You don't need to write a `parse_duration` function: the `humantime` library provides one:

```
#[structopt(..., parse(try_from_str = humantime::parse_duration))]
```

Reply:
Thanks for pointing this out, this PR is much simpler now.